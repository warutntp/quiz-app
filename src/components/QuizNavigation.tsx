import React from "react";

interface QuizNavigationProps {
  questionIndex: number;
  shuffledQuestions: number;
  answers: string[];
  handleNext: () => void;
  handleBack: () => void;
  handleSubmitQuiz: () => void;
}

const QuizNavigation: React.FC<QuizNavigationProps> = ({
  questionIndex,
  shuffledQuestions,
  answers,
  handleNext,
  handleBack,
  handleSubmitQuiz,
}) => {
  return (
    <div
      className={`flex mt-4 ${
        questionIndex > 0 ? "justify-between" : "justify-end"
      }`}
    >
      {questionIndex > 0 && (
        <button
          onClick={handleBack}
          className="py-2 px-4 bg-gray-500 text-white rounded hover:bg-gray-700"
        >
          Back
        </button>
      )}
      {questionIndex < shuffledQuestions - 1 && (
        <button
          onClick={handleNext}
          className={`py-2 px-4 ${
            answers[questionIndex] ? "bg-blue-500" : "bg-gray-500"
          } text-white rounded hover:bg-blue-700`}
          disabled={!answers[questionIndex]}
        >
          Next
        </button>
      )}
      {questionIndex === shuffledQuestions - 1 && (
        <button
          onClick={handleSubmitQuiz}
          className={`py-2 px-4 ${
            answers.some((answer) => answer === null)
              ? "bg-gray-500"
              : "bg-green-500"
          } text-white rounded hover:bg-green-700`}
          disabled={answers.some((answer) => answer === null)}
        >
          Submit
        </button>
      )}
    </div>
  );
};

export default QuizNavigation;
