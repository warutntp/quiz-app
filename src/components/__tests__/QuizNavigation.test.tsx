import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import QuizNavigation from "../QuizNavigation";
import "@testing-library/jest-dom/extend-expect";

test("QuizNavigation", () => {
  const handleNext = jest.fn();
  const handleBack = jest.fn();
  const handleSubmitQuiz = jest.fn();

  const { container } = render(
    <QuizNavigation
      questionIndex={0}
      shuffledQuestions={10}
      answers={["Answer"]}
      handleNext={handleNext}
      handleBack={handleBack}
      handleSubmitQuiz={handleSubmitQuiz}
    />
  );
  expect(container).toMatchSnapshot();
});

test("renders navigation buttons and handles clicks correctly", () => {
  const handleNext = jest.fn();
  const handleBack = jest.fn();
  const handleSubmitQuiz = jest.fn();

  render(
    <QuizNavigation
      questionIndex={0}
      shuffledQuestions={10}
      answers={["Answer"]}
      handleNext={handleNext}
      handleBack={handleBack}
      handleSubmitQuiz={handleSubmitQuiz}
    />
  );

  const nextButton = screen.getByText("Next");
  fireEvent.click(nextButton);
  expect(handleNext).toHaveBeenCalled();
});
