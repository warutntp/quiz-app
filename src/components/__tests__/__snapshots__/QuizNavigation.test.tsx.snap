// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizNavigation 1`] = `
<div>
  <div
    class="flex mt-4 justify-end"
  >
    <button
      class="py-2 px-4 bg-blue-500 text-white rounded hover:bg-blue-700"
    >
      Next
    </button>
  </div>
</div>
`;
